File: node/src/resources/webhooks.ts
import { TransmitClient } from '../client';
import { CreateWebhookRequest } from '../types';

export class Webhooks {
  private client: TransmitClient;

  constructor(client: TransmitClient) {
    this.client = client;
  }

  async list(): Promise<any[]> {
    return this.client.get<any[]>('/api/v1/developers/webhooks');
  }

  async create(data: CreateWebhookRequest): Promise<any> {
    return this.client.post<any>('/api/v1/developers/webhooks', data);
  }

  async retrieve(id: string): Promise<any> {
    return this.client.get<any>(`/api/v1/developers/webhooks/${id}`);
  }

  async update(id: string, data: Partial<CreateWebhookRequest>): Promise<any> {
    return this.client.put<any>(`/api/v1/developers/webhooks/${id}`, data);
  }

  async delete(id: string): Promise<void> {
    return this.client.delete<void>(`/api/v1/developers/webhooks/${id}`);
  }

  async getEvents(id: string): Promise<any[]> {
    return this.client.get<any[]>(`/api/v1/developers/webhooks/${id}/events`);
  }
}

File: rust/src/resources/webhooks.rs
use crate::client::{Client, TransmitError};
use serde_json::Value;

pub struct Webhooks<'a> {
    client: &'a Client,
}

impl<'a> Webhooks<'a> {
    pub fn new(client: &'a Client) -> Self {
        Self { client }
    }

    pub async fn list(&self) -> Result<Vec<Value>, TransmitError> {
        self.client.get("/api/v1/developers/webhooks").await
    }

    pub async fn create(&self, body: &Value) -> Result<Value, TransmitError> {
        self.client.post("/api/v1/developers/webhooks", body).await
    }

    pub async fn retrieve(&self, id: &str) -> Result<Value, TransmitError> {
        self.client.get(&format!("/api/v1/developers/webhooks/{}", id)).await
    }

    pub async fn delete(&self, id: &str) -> Result<Value, TransmitError> {
        self.client.delete(&format!("/api/v1/developers/webhooks/{}", id)).await
    }
}
